kore/src/Kore/Step/Rule/Expand.hs

```diff
@@ -79,12 +82,14 @@ import Kore.Variables.UnifiedVariable
     , extractElementVariable
     )
 
-expandOnePathSingleConstructors
-    :: SmtMetadataTools attributes
-    -> OnePathRule Variable
-    -> OnePathRule Variable
-expandOnePathSingleConstructors metadataTools (OnePathRule rule) =
-    OnePathRule (expandSingleConstructors metadataTools rule)
+expandClaimSingleConstructors
+    :: Coercible (RulePattern Variable) claim
+    => Coercible claim (RulePattern Variable)
+    => SmtMetadataTools attributes
+    -> claim
+    -> claim
+expandClaimSingleConstructors metadataTools claim =
+    coerce (expandSingleConstructors metadataTools (coerce claim))
```

Comment:
When we have a function applied over a coercion like this, it makes me think this would be better in the class. Could we lift this to the class? Or going even further, make this a strategy step, so that we expand any single-constructor variables at the start of each proof step? Or going perhaps to far: make this a step in the variable simplifier? :sunglasses: 

That last is probably too far because of the quantification problem, but I think we could generalize this a bit.